domain/src/main/kotlin/no/nav/su/se/bakover/domain/beregning/fradrag/FradragStrategy.kt

```diff
@@ -1,40 +1,139 @@
 package no.nav.su.se.bakover.domain.beregning.fradrag
 
+import arrow.syntax.function.pipe
 import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.domain.Minstepensjonsnivå
+import no.nav.su.se.bakover.domain.beregning.Sats
 
 internal sealed class FradragStrategy {
-    abstract fun beregnFradrag(forventetInntekt: Int, fradrag: List<Fradrag>, periode: Periode): List<Fradrag>
+    fun beregn(fradrag: List<Fradrag>): List<Fradrag> {
+        validate(fradrag)
+        return beregnFradrag(fradrag)
+    }
+
+    protected open fun validate(fradrag: List<Fradrag>) {
+        require(fradrag.harNøyaktigEnForventetInntektFor(FradragTilhører.BRUKER)) { "Fradrag må inneholde brukers forventede inntekt etter uførhet." }
```

Comment:
Hvis jeg forstår det rett, bør den første her føre til en http statuskode 500, mens de 2 andre bør føre til 400. Og loggnivået bør kanskje og være forskjellig? Siden noe vil være programmeringsfeil og andre ting vil være "brukerfeil"?

Bryr vi oss om å skille på dette i frontend? Eller har vi god nok validering frontend?

Reply:
Dette er et godt poeng som vi nok bør ta stilling til når vi begynner å se på koblingen mot frontend. Har ikke tatt høyde for eventuelle behov derfra enda, men er enig med deg at det fremstår som at dette potensielt bør føre til ulike httpkoder.